client/services/code_evaluators/PythonPrereqCheckServiceSpec.js

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the PythonPrereqCheckService.
+ */
+
+describe('PythonPrereqCheckService', function() {
+	var PythonPrereqCheckService;
+
+	beforeEach(module('tie'));
+	beforeEach(inject(function($injector) {
+		PythonPrereqCheckService = $injector.get(
+			'PythonPrereqCheckService');
+	}));
+
+	describe('checkStarterCodePresent', function() {
+		var starterCode = ['def myFunction(arg):\n',
+			'\treturn result\n'].join('');
+
+		it('returns true if starter code lines are found in code', function() {
+			var starterCode = ['def myFunction(arg):\n',
+			'\treturn result\n'].join('');
+			var code = ['def myFunction(arg):\n',
+			'\tresult = arg.rstrip()\n',
+			'\treturn result\n'].join('');
+			var starterCodePresent =
+				PythonPrereqCheckService.checkStarterCodePresent(
+					starterCode, code);
+			expect(starterCodePresent).toEqual(true);
+		});
+		it('returns false if starter code line is not found (modified)',
+			function() {
+				var code = ['def yourFunction(arg):\n',
+				'\tresult = arg.rstrip()\n',
+				'\treturn result\n'].join('');
+				var starterCodePresent =
+					PythonPrereqCheckService.checkStarterCodePresent(
+						starterCode, code);
+				expect(starterCodePresent).toEqual(false);
+		});
+		it('returns false if starter code line is not found (deleted)',
+			function() {
+				var code = ['\tresult = arg.rstrip()\n',
+				'\treturn result\n'].join('');
+				var starterCodePresent =
+					PythonPrereqCheckService.checkStarterCodePresent(
+						starterCode, code);
+				expect(starterCodePresent).toEqual(false);
+		});
```

Comment:
I mentioned before that the test "it returns false when starter code lines are present, but out of order" seems odd, but I should have been clearer. I think this is a good test to include here, but we should instead check that it returns `true`.